### DISCUSSION TITLE
Add text to existing text through plugin, possible?

### DISCUSSION THREAD

**Discussion by MiltonBalaOfficial** [xournalpp/xournalpp#6100] [Q&A]
Is it possible to add text to an existing text through plugin?

**Comment by rolandlo** [comment]
You can get the text of some text element, remove the text element and add another text element with some more text added to it.

**Reply by MiltonBalaOfficial** [reply]
Thanks for the reply. Actually I do the same thing to insert superscript or subscript when I type using text tool. But sometimes (when I wanted to modify my previous typed text, or when I wanted to edit text copied from web as the Superscript or subscript don't remain as it is) I need to insert text in the middle of typed text. Is there any solution to my problem?

**Reply by MiltonBalaOfficial** [reply]
> I need to insert text in the middle of typed text. Is there any solution to my problem?

I have found a solution, If we type `any symbol that is not used for normal text` for example `backtick`, at the position where we want to insert the `desired symbol` and then the  `backtick` can be replaced by the desired `text symbol`. I am providing my code bellow, if anyone wants to suggest anything, please do. 


function textHelp(textSymbol)
    -- Trigger the rectangle selection tool for selecting the working textbox
    app.uiAction({ action = "ACTION_TOOL_SELECT_RECT" })

    -- Get the currently selected text table
    local selectedTexts = app.getTexts("selection")

    -- Use the first selected text as a reference
    local firstText = selectedTexts[1]
    
--(User has to add a backtick (`) where he wants the symbol get inserted, if no backtick then symbol will be added at the last)
    -- Replace backtick (`) with textSymbol if found 
    if firstText.text:find("`") then
        finalText = firstText.text:gsub("`", textSymbol)
    else
        -- If backtick not found, append textSymbol to the string
        finalText = firstText.text .. textSymbol
    end

    -- Add a new text object with the updated content
    app.addTexts {
        texts = {{
                text = finalText,
                font = {
                    name = firstText.font.name,
                    size = firstText.font.size
                    },
                color = firstText.color,
                x = firstText.x,
                y = firstText.y
                }},
        allowUndoRedoAction = "individual"
    }

    -- Delete the old text object and refresh the UI
    app.uiAction({ action = "ACTION_DELETE" })
    app.refreshPage()

    -- Switch to the text tool so that could be continued
    app.uiAction({ action = "ACTION_TOOL_TEXT" })
end
